### PR TITLE
fix(profile): harden playlist discovery fetcher

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary
- add backoff and quieter handling for expected Spotify page failures
- escape quoted artist names before Google CSE queries
- catch Google CSE failures in discovery and cover the hardening paths with tests

## Verification
- doppler run --project jovie-web --config dev -- pnpm --filter web exec vitest run tests/unit/profile/featured-playlist-fallback-discovery.test.ts
- doppler run --project jovie-web --config dev -- ./node_modules/.bin/tsc --noEmit
- pnpm biome check --write apps/web
- git push origin itstimwhite/playlist-fallback-discovery-fetch (pre-push passed)

## Summary by CodeRabbit

* **New Features**
  * Added enhanced playlist discovery mechanism to identify and validate featured playlists using search and direct validation.

* **Tests**
  * Added comprehensive test coverage for playlist discovery functionality, including success scenarios, error handling, and edge cases.

### CODE REVIEW

File: apps/web/lib/profile/featured-playlist-fallback-discovery.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+export async function discoverThisIsPlaylistCandidate(input: {
+  readonly artistName: string;
+  readonly artistSpotifyId: string;
+}): Promise<FeaturedPlaylistFallbackCandidate | null> {
+  const artistName = input.artistName.trim();
+  const safeArtistName = artistName.replaceAll('"', '');
+  const artistSpotifyId = input.artistSpotifyId.trim();
+
+  if (!artistName || !artistSpotifyId) {
+    return null;
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/lib/profile/featured-playlist-fallback-discovery.ts:92]: Fixed in 0c10d5a147537ff14fecd09707abbbc19a0ed1a9 — added `!safeArtistName` to the early return guard so all-quotes artist names return null immediately.

File: apps/web/lib/profile/featured-playlist-fallback-discovery.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      });
+
+      if (response.ok) {
+        return await response.text();
+      }
+
+      if (response.status >= 500 && attempt < MAX_PAGE_FETCH_ATTEMPTS) {
+        await sleep(RETRY_BACKOFF_BASE_MS * attempt);
+        continue;
+      }
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/lib/profile/featured-playlist-fallback-discovery.ts]: Fixed in 0c10d5a147537ff14fecd09707abbbc19a0ed1a9 — restructured the 5xx block to `return null` on the final attempt instead of falling through to the generic warning+return path.

File: apps/web/lib/profile/featured-playlist-fallback-discovery.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  readonly artistSpotifyId: string;
+}): Promise<FeaturedPlaylistFallbackCandidate | null> {
+  const artistName = input.artistName.trim();
+  const safeArtistName = artistName.replaceAll('"', '');
+  const artistSpotifyId = input.artistSpotifyId.trim();
+
+  if (!artistName || !artistSpotifyId) {
+    return null;
+  }
+
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/lib/profile/featured-playlist-fallback-discovery.ts:94]: Fixed in 0c10d5a147537ff14fecd09707abbbc19a0ed1a9 — added `!safeArtistName` to the guard. Now the function returns null early when `artistName` consists entirely of quotes.
